src/main/scala/epic/dense/LowRankQuadraticTransform.scala

```diff
@@ -0,0 +1,142 @@
+package epic.dense
+
+import breeze.linalg._
+import breeze.linalg.operators.OpMulMatrix
+import epic.features.SegmentedIndex
+import epic.framework.Feature
+import breeze.util.Index
+
+import scala.runtime.ScalaRunTime
+import scala.util.Random
+
+case class LowRankQuadraticTransform[FV](numOutputs: Int, numRanks: Int, numLeftInputs: Int, numRightInputs: Int, innerTransform: Transform[FV, DenseVector[Double]]) extends OutputTransform[FV, DenseVector[Double]] {
```

Comment:
I think I found it to be slightly beneficial compared to just linear, but not better than a neural net and much slower; even with a pretty low-rank matrix you end up with way more arithmetic operations so I didn't end up using it.